### PR TITLE
feat(collector-watcher): discover and store component READMEs

### PR DESCRIPTION
**Pull request by MeloveGupta**
## What changed

Collector-watcher now looks for a README.md next to each component it already scans, and if one exists, then stores it content-addressed under ```component_readmes/``` (same idea as how java-instrumentation-watcher stores library readmes, just adapted since collector has two distributions - core and contrib, instead of just one version key).

## Why

Part 1/3 of #762 

Readmes on collector components have useful context that's currently invisible in the explorer. This PR just makes the content available and stored properly.

## How it was tested

I ran the full collector-watcher suite, 180 passed (156 existing + 24 new). So for the two trickier bits I actually broke the code on purpose first: removed the try/except around the readme-saving step and confirmed the "doesn't block the sync" test actually fails without it, then put it back. I also had a wrong assumption in one of my own tests about whether an empty component_readmes dir gets created when there's nothing to save, caught that by actually running it against the real behavior, and fixed the test to match.

## Is this a breaking change?

No

## Special notes for your reviewer

A couple things:

- Went with porting the pattern into collector's own InventoryManager instead of generalizing BaseInventoryManager, since collector's (distribution, version) key doesn't fit the base class's current single-key design. Generalizing that is being tracked as a separate followup instead of blocking this
- Readme saving is wrapped in a try/except so it can never fail the actual component inventory write, same "never fail the build" idea already used for markdown publishing on the db-builder side
- Named the storage dir ```component_readmes``` instead of reusing java's ```library_readmes```, just to match collector's own naming.

## Checklist

- [x] Tests added or updated for new behavior
- [x] Screenshots attached for any UI changes
- [x] I wrote this PR description myself (AI tools must not check this box on behalf of the author)

### CODE REVIEW

File: ecosystem-automation/collector-watcher/src/collector_watcher/collector_sync.py [hunk 1]
```diff
@@ -190,6 +196,17 @@ def save_version(
...
+        try:
+            readmes = discover_component_readmes(repo_path, components)
+            written = self.inventory_manager.save_component_readmes(distribution, version, readmes.items())
+            if written:
+                logger.info("  Saved %d component README(s)", written)
+        except OSError as e:
+            # README publishing is best-effort and must never fail the sync -
+            # the component inventory itself is the critical data.
+            logger.warning("  Failed to save component READMEs for %s %s: %s", distribution, version, e)
+
```
[the remark this answers is not in the document]
Comment (MeloveGupta) [reply ecosystem-automation/collector-watcher/src/collector_watcher/collector_sync.py:229]: I did tested it, a crash right at save_versioned_inventory left version_exists() returning True with zero actual component data, which means process_latest_release would skip that version forever thinking it's already synced. I have swapped the order so the real inventory write happens first, readme saving after. Also I have added two more regression tests, one for a crash before the write (should leave no trace) and one for a crash during readme saving after the write succeeds (should be harmless). I've pushed the fix.

File: ecosystem-automation/collector-watcher/tests/test_collector_sync.py [hunk 2]
```diff
@@ -566,3 +566,46 @@ def test_baseline_not_updated_for_prereleases(collector_sync):
...
+
+    readme_map = collector_sync.inventory_manager.load_component_readme_map("core", version)
+    assert "otlpreceiver" in readme_map
+    content = collector_sync.inventory_manager.load_component_readme_content(
+        "core", version, "otlpreceiver", readme_map["otlpreceiver"]
+    )
+    assert content == "# OTLP Receiver"
+
+
+def test_save_version_with_no_readmes_writes_no_readme_dir(collector_sync, sample_components, temp_inventory_dir):
```
[the remark this answers is not in the document]
Comment (MeloveGupta) [reply ecosystem-automation/collector-watcher/tests/test_collector_sync.py]: I have renamed it to test_save_version_with_no_readmes_persists_no_readme_content, now it matches what it actually checks now. I have pushed the fix.

**Review by vitorvasc — APPROVED** [review]
